test/install/shamefullyFlatten.ts

```diff
@@ -112,3 +113,88 @@ test('--no-shamefully-flatten throws exception when executed on node_modules ins
     t.ok(err.message.indexOf('This node_modules was installed with --shamefully-flatten option.') === 0)
   }
 })
+
+test('flatten by alias', async (t: tape.Test) => {
+  const project = prepare(t)
+
+  // pkg-with-1-aliased-dep aliases dep-of-pkg-with-1-dep as just "dep"
+  await installPkgs(['pkg-with-1-aliased-dep'], await testDefaults({shamefullyFlatten: true}))
+
+  await project.has('pkg-with-1-aliased-dep')
+  await project.has('dep')
+  await project.hasNot('dep-of-pkg-with-1-dep')
+
+  const modules = await project.loadModules()
+  t.deepEqual(modules.hoistedAliases, {'localhost+4873/dep-of-pkg-with-1-dep/100.1.0': [ 'dep' ]}, '.modules.yaml updated correctly')
+})
+
+test('should remove aliased flattened dependencies', async function (t) {
+  const project = prepare(t)
+
+  await installPkgs(['pkg-with-1-aliased-dep'], await testDefaults({shamefullyFlatten: true}))
+  await uninstall(['pkg-with-1-aliased-dep'], await testDefaults({shamefullyFlatten: true}))
+
+  await project.hasNot('pkg-with-1-aliased-dep')
+  await project.hasNot('dep-of-pkg-with-1-dep')
+  let caught = false
+  try {
+    await resolveLinkTarget('./node_modules/dep')
+  } catch (e) {
+    caught = true
+  }
+  t.ok(caught, 'dep removed correctly')
```

Comment:
You could use `await project.hasNot('dep')`, right?

Reply:
No I couldn't, because it would pass if the link `dep` was still there.